implementations/rust/ockam/ockam_api/src/cli_state/vaults.rs

```diff
@@ -364,8 +364,10 @@ impl NamedVault {
         }
     }
 
-    async fn database(&self) -> Result<Arc<SqlxDatabase>> {
-        Ok(Arc::new(SqlxDatabase::create(self.path.as_path()).await?))
+    async fn database(&self) -> Result<SqlxDatabase> {
+        // FIXME: We should really have one instance of the SqlxDatabase per process, also the node_name is incorrect,
+        //  but we don't use it yet for the vault storage
+        Ok(SqlxDatabase::create(self.path.as_path(), "vault_node_name".to_string()).await?)
```

Comment:
Perhaps having a `OnceLock` static + adding a `SqlxDatabase::new` method would do the trick?

Reply:
I think that `SqlxDatabase` shouldn't be created here at all. It should be created once per db file / per process (and we currently have only one db file). Then that instance can be cloned and used everywhere else. But this has nothing to do with this PR

Reply:
Maybe we could achieve that with OnceLock inside the SqlxDatabase, but I was thinking we could achieve the same by choosing the correct place where we instantiate the object